src/organisation/components/pba-number-input/pba-number-input.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { AbstractControl, FormControl, FormGroup } from '@angular/forms';
+import { ErrorMessagesModel } from '@hmcts/rpx-xui-common-lib/lib/gov-ui/models';
+
+@Component({
+    selector: 'app-prd-pba-number-input-component',
+    templateUrl: './pba-number-input.component.html',
+})
+export class PbaNumberInputComponent implements OnInit {
+
+    static PBA_GENERIC_ERROR_MESSAGE = ['There is a problem. Enter a PBA number, for example PBA1234567'];
+    static PBA_UNIQUE_ERROR_MESSAGE = ['You have entered this PBA number more than once'];
+    static PBA_EXISTING_ERROR_MESSAGE = ['This PBA number is already associated with your organisation'];
+
+    @Input() public id: string;
+    @Input() public group: FormGroup;
+
+    public control: FormControl;
+
+    public errorMessages: ErrorMessagesModel  = {
+        isInvalid: false,
+        messages: []
+    };
+
+    ngOnInit() {
+        this.group.valueChanges.subscribe(() => this.onPbaNumberValueChange());
+    }
+
+    private onPbaNumberValueChange(): void {
+        const control = this.group.get('pbaNumber');
+        control.patchValue(this.formatPbaNumber(control), { emitEvent: false });
```

Comment:
Are we guaranteed to have a `pbaNumber` control within this?

Reply:
Yes, it should do, I could add a check for this, but what would you propose? Throwing an error if it does not exist?

Reply:
added a check